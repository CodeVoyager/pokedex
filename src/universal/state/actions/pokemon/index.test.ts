/* eslint-disable @typescript-eslint/no-explicit-any */
import {
  setPokemonListPageAction,
  setPokemonListAction,
  setPokemonAction,
} from '.';

describe('State::Actions', () => {
  describe('Pokemon', () => {
    describe('setPokemonListPageAction creator', () => {
      test('should return correct action', () => {
        expect(setPokemonListPageAction(1)).toMatchSnapshot();
      });
    });
    describe('setPokemonListAction creator', () => {
      test('should return correct action', () => {
        expect(setPokemonListAction([])).toMatchSnapshot();
      });
    });
    describe('setPokemonAction creator', () => {
      test('should return correct action', () => {
        expect(
          setPokemonAction({
            pokemon: true,
          } as any)
        ).toMatchSnapshot();
      });
    });
  });
});
