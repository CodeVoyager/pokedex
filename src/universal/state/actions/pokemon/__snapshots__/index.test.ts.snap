// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`State::Actions Pokemon setPokemonAction creator should return correct action 1`] = `
Object {
  "payload": Object {
    "pokemon": true,
  },
  "type": "SET_POKEMON",
}
`;

exports[`State::Actions Pokemon setPokemonListAction creator should return correct action 1`] = `
Object {
  "payload": Array [],
  "type": "SET_POKEMON_LIST",
}
`;

exports[`State::Actions Pokemon setPokemonListPageAction creator should return correct action 1`] = `
Object {
  "payload": 1,
  "type": "SET_POKEMON_LIST_PAGE",
}
`;
